### ISSUE TITLE
Make pipeline parallelism modular

### ISSUE THREAD

**Issue by siddharth9820** [hpcgroup/axonn#42]
- [x] Remove the need to rely on AxoNN for mixed precision. User should supply an autocast context. (https://github.com/karpathy/nanoGPT/blob/master/train.py#L112)
- [x] create dataloader - add option to use one of gradient accumulation as well as microbatch size
